src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -66,87 +63,29 @@
   @Override
   public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgumentException {
     super.configurePipeline(configurer);
-    try {
-      schema = Schema.parseJson(config.schema);
-    } catch (IOException e) {
-      throw new IllegalArgumentException(
-        String.format("Unable to parse output schema. Reason: %s", e.getMessage()), e
-      );
-    }
-
-    if (!config.containsMacro("field")) {
-      Schema.Field field = schema.getField("field");
-      if (field != null && field.getSchema().getType() == Schema.Type.BYTES) {
-        throw new IllegalArgumentException(
-          String.format(
-            "Field '%s' should be a byte array.", field.getName()
-          )
-        );
+    Schema inputSchema = configurer.getStageConfigurer().getInputSchema();
+    // If a inputSchema and audioField is available then verify that the schema does contain the given audioField
+    // and that the type is byte array. This will allow to fail fast i.e. during deployment time.
+    if (inputSchema != null && !config.containsMacro("audiofield")) {
+      Schema.Field field = inputSchema.getField(config.audioField);
+      if (field != null && field.getSchema().getType() != Schema.Type.BYTES) {
```

Comment:
this is incorrect, as it doesn't handle nullables correctly. If it's nullable bytes, it will be a union. Need to do something like:
```
Schema fieldSchema = field.getSchema();
Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
```